lib/std/sort/block.zig

```diff
@@ -103,15 +103,17 @@ pub fn block(
     context: anytype,
     comptime lessThanFn: fn (@TypeOf(context), lhs: T, rhs: T) bool,
 ) void {
+    // const is_ctx_comptime = @typeInfo(@TypeOf(lessThanFn)).Fn.Params[0].is_comptime;
     const lessThan = if (builtin.mode == .Debug)
-        if (@inComptime()) struct {
-            fn lessThan(comptime ctx: @TypeOf(context), lhs: T, rhs: T) bool {
-                const lt = lessThanFn(ctx, lhs, rhs);
-                const gt = lessThanFn(ctx, rhs, lhs);
-                std.debug.assert(!(lt and gt));
-                return lt;
-            }
-        }.lessThan else struct {
+        // if (is_ctx_comptime) struct {
+        //     fn lessThan(comptime ctx: @TypeOf(context), lhs: T, rhs: T) bool {
+        //         const lt = lessThanFn(ctx, lhs, rhs);
+        //         const gt = lessThanFn(ctx, rhs, lhs);
+        //         std.debug.assert(!(lt and gt));
+        //         return lt;
+        //     }
+        // }.lessThan else
```

Comment:
Why have you commented this logic out? Is it not the whole point of this PR?

Reply:
It's added back in another patch.